src/ui/DynamicFacet/DynamicFacetRange.ts

```diff
@@ -49,16 +49,27 @@ export class DynamicFacetRange extends DynamicFacet implements IComponentBinding
       localizedString: () => l('To'),
       section: 'CommonOptions'
     }),
+    field: ComponentOptions.buildFieldOption({ required: true, section: 'CommonOptions' }),
     /**
      * The string format to apply to the minimum and maximum value of each range displayed in the facet.
      *
-     * **Default:** [`number`]{@link DynamicFacetRangeValueFormat.number}
+     * **Default:** If the [`field`]{@link DynamicFacet.options.field} contains "date", the format will be [`date`]{@link DynamicFacetRangeValueFormat.date}.
+     * Else, it will be [`number`]{@link DynamicFacetRangeValueFormat.number}.
      *
      * @examples date
      */
     valueFormat: ComponentOptions.buildStringOption({
-      defaultValue: DynamicFacetRangeValueFormat.number,
-      postProcessing: value => (isFacetRangeValueFormat(value) ? value : DynamicFacetRangeValueFormat.number),
+      postProcessing: (value, options: IDynamicFacetRangeOptions) => {
+        if (!value && options.field.indexOf('date') !== -1) {
+          return DynamicFacetRangeValueFormat.date;
+        }
+
+        if (isFacetRangeValueFormat(value)) {
```

Comment:
I think we should do this check first before `options.field.indexOf('date') !== -1` to prioritize a valid user-specified value rather than the field name.

For example, say the user uses a field called `mandateNumber`. They might want to format the values as `number`.
However, because `date` is in the field name, we will override their preference and use `date`.

We should also add a test that captures this case of interplay between the two options.

Reply:
👍